### DISCUSSION TITLE
Slow compilation due to macros?

### DISCUSSION THREAD

**Discussion by PatrickDanino** [pointfreeco/swift-perception#50] [Q&A] [answered]
We noticed that our project has been taking considerably longer to compile. We stumbled on this [thread](https://forums.swift.org/t/compilation-extremely-slow-since-macros-adoption/67921/68), which appears to suggest that it could be due to using macros.

Any thoughts to a potential workaround?

**Comment by stephencelis** [comment, marked as answer]
This seems to be the cost of macros at the moment. In the thread you link a core team member is quoted to knowing that there is an issue, so all we can hope for is for Apple to address it in the future.

**Comment by PatrickDanino** [comment]
In case it helps anyone, we experimented with a [pre-compiled Swift Syntax framework](https://github.com/sjavora/swift-syntax-xcframeworks). Not only does that appear to work, it cut down our rebuild time from ~120s to ~10s. Each and every time we rebuild.
